packages/plugin-core/src/views/NativeTreeView.ts

```diff
@@ -68,17 +68,32 @@ export class NativeTreeView implements Disposable {
     >;
 
     result.then(() => {
-      this.treeView = window.createTreeView(DendronTreeViewKey.TREE_VIEW, {
+      const treeView = window.createTreeView(DendronTreeViewKey.TREE_VIEW, {
         treeDataProvider,
         showCollapseAll: true,
       });
 
+      treeView.onDidChangeVisibility(async (e) => {
+        // when the tree view is first shown ever during the session,
+        // VSCode fires a tree view `.reveal` internally to initially load in the nodes.
+        // when this is happening, if we trigger a `reveal` again, there will be an error.
+        // onDidChangeVisibility doesn't seem to wait for the reveal to finish, so
+        // we have to waitfor the initial load to finish before triggering `reveal`.
+        // It seems like setting a very short timeout is enough to this cooperate with VSCode.
+        // TODO: Remove timeout once this bug is resolved.
```

Comment:
A lot of head-scratching on this one 😄 I'll submit a bug report to VSCode for this and come back to fix this once that's resolved.

Reply:
But is it possible that the intended behavior is for visibility to be set to true _then_ data loading happens?

Reply:
that makes sense, but there is no safeguard here for the extension authors that want to trigger `.reveal` on the tree view. nor can I find any reference of a way to wait for that internal call before we go ahead and trigger it ourselves.

Reply:
If we were to handle only what we are explicitly triggering, we would have a queue that lets us wait for the previous call to resolve (recommended here by the team: https://github.com/microsoft/vscode/issues/142328#issuecomment-1046907141). but the initial call that VSCode makes to populate the tree view on the first `visible: true` we have no info on the status as-is.

Reply:
Agree with your points - main point I was trying to make is that I'm not sure what kind of 'bug' per se to file against VS Code.  Perhaps an API request to signal when initial rendering is completed? But I'm not sure if that's the design they want.  It's worth asking at the very least

Reply:
Agreed. that's the plan.